src/update-repo-task-status.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  AnalysisStatus,
+  getRepoTask,
+  setVariantAnalysisCanceled,
+  setVariantAnalysisFailed,
+} from "./gh-api-client";
+import {
+  getControllerRepoId,
+  getRepos,
+  getVariantAnalysisId,
+  getWorkflowStatus,
+} from "./inputs";
+
+// All states that indicate the repository has been scanned and cannot
+// change status anymore.
+const COMPLETED_STATES = ["succeeded", "failed", "canceled", "timed_out"];
+
+/**
+ * If the overall variant analysis workflow failed or was canceled,
+ * propagate the failure/cancellation status to the individual repo tasks.
+ */
+async function updateRepoTaskStatus(): Promise<void> {
+  const controllerRepoId = getControllerRepoId();
+  const variantAnalysisId = getVariantAnalysisId();
+  const workflowStatus = getWorkflowStatus();
+
+  if (!controllerRepoId || !variantAnalysisId || !workflowStatus) {
```

Comment:
I don't think this is actually possible: if any of these are not passed, the `getInput` function should already throw. The types also don't specify that these can be `undefined`.

Reply:
I think you're right ✔️  Removed that block!